### PR TITLE
add UnlockAndRelock

### PR DESCRIPTION
**Pull request by micbar**
# Description

Implements https://github.com/cs3org/wopiserver/issues/86

### CODE REVIEW

**Comment by micbar** [comment]
@glpatcern

File: cs3/storage/provider/v1beta1/provider_api.proto [hunk 1]
```diff
@@ -1022,6 +1030,33 @@ message RefreshLockResponse {
...
+message UnlockAndRelockResponse {
+  // REQUIRED.
+  // The response status.
+  cs3.rpc.v1beta1.Status status = 1;
+  // OPTIONAL.
+  // Opaque information.
+  cs3.types.v1beta1.Opaque opaque = 2;
+  // REQUIRED.
+  // The lock metadata
+  Lock lock = 3;
```
Comment (glpatcern) [inline cs3/storage/provider/v1beta1/provider_api.proto:1057]: Why would you need to return the new lock data structure on response? After all this operation is similar to RefreshLock, the server could just say "done" or "failed".

Comment (micbar) [reply cs3/storage/provider/v1beta1/provider_api.proto:1057]: ok, you are correct. The new refresh metadata is already available for the caller. will change.

**Comment by glpatcern** [comment]
@micbar I managed to check the current behavior of Reva `master` with the eosfs storage driver, and in fact the `RefreshLock` implementation does incorporate the `UnlockAndRelock` case. After all, the only behavioral difference between the two is enforcing or not the check on the `lock_id` payload. The decomposedfs implementation does enforce it:

https://github.com/cs3org/reva/blob/edge/pkg/storage/utils/decomposedfs/node/locks.go#L179-L182

And the eosfs implementation does not (similarly to the xroot "eos native" legacy implementation of the wopiserver). This is why for us the Microsoft WOPI validator test suite always passed.

With those premises, I'm really not convinced that it makes sense to add an extra call to the CS3APIs with such a minor behavioral difference, given that:
* the current specs do require to check the lock holder and the expiration time to protect the `RefreshLock` (as well as the `Unlock`) call
* the specs do not require to enforce that check on the opaque `lock_id` payload

Technically, dropping those 3 lines in locks.go and reverting cs3org/wopiserver#85 will make you pass the tests, with full atomicity preserved, instead of the overhead of adding a new API where the implementation would need to be a clone of the existing `RefreshLock` code - except for those 3 lines - for all storage providers.

I'd like to have more voices from @butonic and @labkode prior to go on or not.

**Comment by micbar** [comment]
fine for me. I was also looking into the implementation.

**Pull request closed without merging**